Source/Core/Common/StringUtil.cpp

```diff
@@ -425,44 +427,68 @@ std::string UTF16ToUTF8(const std::wstring& input)
 
 std::wstring CPToUTF16(u32 code_page, const std::string& input)
 {
-  auto const size = MultiByteToWideChar(code_page, 0, input.data(), (int)input.size(), nullptr, 0);
+  auto const size =
+      MultiByteToWideChar(code_page, 0, input.data(), static_cast<int>(input.size()), nullptr, 0);
 
   std::wstring output;
   output.resize(size);
 
   if (size == 0 ||
-      size != MultiByteToWideChar(code_page, 0, input.data(), (int)input.size(), &output[0],
-                                  (int)output.size()))
+      size != MultiByteToWideChar(code_page, 0, input.data(), static_cast<int>(input.size()),
+                                  &output[0], static_cast<int>(output.size())))
   {
     output.clear();
   }
 
   return output;
 }
 
+std::string UTF16ToCP(u32 code_page, const std::wstring& input)
+{
+  auto const size = WideCharToMultiByte(code_page, 0, input.data(), static_cast<int>(input.size()),
+                                        nullptr, 0, nullptr, false);
+
+  std::string output;
+  output.resize(size);
+
+  if (size == 0 ||
+      size != WideCharToMultiByte(code_page, 0, input.data(), static_cast<int>(input.size()),
+                                  &output[0], static_cast<int>(output.size()), nullptr, false))
+  {
+    output.clear();
+    ERROR_LOG(COMMON, "WCTMB Error in String '%s': %d", input.c_str(), GetLastError());
```

Comment:
```GetLastError()``` must be the first thing you call, even if you have to create another variable to store the result. We already had weird behaviours with ```WSAGetLastError()``` in the past (https://github.com/dolphin-emu/dolphin/pull/3143) and I don't want that to happen again.

Quoted from the [documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/ms679360(v=vs.85).aspx):
> **Remarks**
>
> Functions executed by the calling thread set this value by calling the **SetLastError** function. You should call the **GetLastError** function immediately when a function's return value indicates that such a call will return useful data. That is because some functions call **SetLastError** with a zero when they succeed, wiping out the error code set by the most recently failed function.